src/server/journal/journal_slice.cc

```diff
@@ -118,9 +118,25 @@ error_code JournalSlice::Close() {
 void JournalSlice::AddLogRecord(const Entry& entry) {
   DCHECK(ring_buffer_);
 
+  iterating_cb_arr_ = true;
   for (const auto& k_v : change_cb_arr_) {
+    // Comment this to cause bug.
+    if (k_v.first == 0)
+      continue;
+
+    // Testing details:
+    // Run pytest with many replicas that crash during stable state sync
+    // They are described by the second array:
+    // (8, [], [2] * 16, 10000)
+
+    // (8, [], [2] * 12, 10000) is part of the test and crashes for me
     k_v.second(entry);
```

Comment:
I've left a guide on how to reproduce the ordering bug

Reply:
I did not succeed unfortunately.